### ISSUE TITLE
Potential Improvements to cheftap importer

### ISSUE THREAD

**Issue by ModdedGamers** [vabene1111/recipes#532]
**Is your feature request related to a problem? Please describe.**
The ChefTap importer isn't working properly (always says 0 recipes imported, and does not import recipes.)

**Describe the solution you'd like**
Here are a few potential improvements to the ChefTap importer function: 

1. The recipe name is always the first line, and the URL/Source is the 3rd, unless it was manually typed in, in which case there is no URL on the 3rd line, and it is blank. Wacky, I know. (I understand the ChefTap format is absolute garbage, but I'm trying my best.)
2. Headers can be Title Case, ALL UPPERCASE, or really AnY caSE aT aLL. Headers can also not exist, and in that case the importer should just shove everything into the steps and maybe throw an error to look at a certain recipe(or 50 recipes because the format is crap)
3. It seems that it just pulls the website content with no parsing at all. For instance, look at this monstrosity of a file: [4_Ingredient_Roasted_Butternut_Squash_Soup.txt](https://github.com/vabene1111/recipes/files/6274445/4_Ingredient_Roasted_Butternut_Squash_Soup.txt). What is this crap.
4. In some cases, the header's don't exist, but the recipe format is "simple". For instance, this [apple pie recipe.](https://github.com/vabene1111/recipes/files/6274460/All_American_Apple_Pies.txt) Recipe name, then URL, the ingredients in 1 block, then the steps, and finally the cooking time. However, many recipes are pulled from random websites, and this well-formatted recipe is from the Washington Post, so don't expect greatness out of every recipe.

**Describe alternatives you've considered**
N/A

**Additional context**
running version 0.15.1, in docker for mac. Thank you for your hard work, and if I come up with any other stuff, I'll leave it here.

**Comment by ModdedGamers** [comment]
Also, you may want to consider trying to re-scrape the site from the url that may or may not be on the 3rd line of the file.

**Comment by vabene1111** [comment]
thanks for the feedback! 

The issue you describe is interesting because i tested cheftap with test data exported from my test account worked. 

Could you send me the files you are trying to import (or some example files) that dont work so i can see what is broken ?

**Comment by ModdedGamers** [comment]
Well, I linked 2 of the weird files, but I can send some more weird ones to test the parser.

**Comment by vabene1111** [comment]
just to make sure, the files you send could not be imported in your instance (following the instructions in the docs) ?

**Comment by ModdedGamers** [comment]
When importing [this zip file](https://github.com/vabene1111/recipes/files/6279399/cheftap_export.zip), after the import is finished, it says "0 recipes imported"

**Comment by vabene1111** [comment]
ok so the problem was that in my test data the txt files were contained in a `cheftap_export` folder inside the zip.

I have changed the discovery algorithm to also find txt files directly in the zip. 

I also changed the structure and imported your recipes and it worked pretty well from what i can see.

this is your zip in the correct structure for 0.15.1, from the next release on your structure should work as well [cheftap_export_test2.zip](https://github.com/vabene1111/recipes/files/6280888/cheftap_export_test2.zip)

Some feedback on if the results were good for you would be great
